paasta_tools/deployd/workers.py

```diff
@@ -103,19 +109,34 @@ def run(self) -> None:
                         max_time=6000,
                     )
                 if bounce_again_in_seconds:
-                    bounce_by = int(time.time()) + bounce_again_in_seconds
-                    service_instance = ServiceInstance(
-                        service=service_instance.service,
-                        instance=service_instance.instance,
-                        bounce_by=bounce_by,
-                        wait_until=bounce_by,
-                        watcher=self.name,
-                        failures=failures,
-                        processed_count=service_instance.processed_count + 1,
-                        bounce_start_time=service_instance.bounce_start_time,
-                        enqueue_time=time.time(),
-                    )
-                    self.instances_to_bounce.put(service_instance)
+                    if failures >= self.max_failures:
+                        self.log.info(
+                            f"{self.name} Worker removing "
+                            f"{service_instance.service}.{service_instance.instance} "
+                            f"from queue because it has failed more than the "
+                            f"maximum of {self.max_failures} times."
+                        )
+                    elif service_instance.processed_count >= self.max_attempts:
+                        self.log.info(
+                            f"{self.name} Worker removing "
+                            f"{service_instance.service}.{service_instance.instance} "
+                            f"from queue because it has been attempted more than "
+                            f"the maximum of {self.max_attempts} times."
```

Comment:
I feel like these should have a more explicit "continue" so that code later down execute? Except I guess it does need to execute. idk.